### DISCUSSION TITLE
Tests run took 5 minutes

### DISCUSSION THREAD

**Discussion by maranmaran** [quii/learn-go-with-tests#803] [Q&A] [answered]
Hi, new to GO :)

All 1172 run for 5 minutes
Is this normal?

Tests seem rather trivial, but the execution time feels so slow for unit tests... 
Is it not parallelized automatically across the test files? 

I understand there's `t.Parallel()`, but I thought that is for tests within the test file specified via code.

I've run more than 3000 tests in 2 minutes in some other frameworks before so this feels wrong to me.
Perhaps I need to tune something in my local setup?

**Comment by quii** [comment, marked as answer]
What command are you running exactly? 

Even in github actions, which will be on shared hardware, the tests for the build take 18s https://github.com/quii/learn-go-with-tests/actions/runs/10356212347/job/28665575241 

![CleanShot 2024-08-20 at 20 51 50](https://github.com/user-attachments/assets/05b25d5f-809c-4483-a17c-b89526d5b4a7)

**Reply by maranmaran** [reply]
Yeah I didn't write that down

`go test .\...` is fast yes.

I've have run my tests via test explorer in vscode that's supported by official go extension, and those are considerably slower.

**Reply by quii** [reply]
Not sure I can help you with that I'm afraid, probably not in scope for this repo, and I don't use vscode :)

**Reply by maranmaran** [reply]
> Not sure I can help you with that I'm afraid, probably not in scope for this repo, and I don't use vscode :) 

Yep all good, I'll close this down
